File: 1241.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int n;
    cin >> n;

    while (n--) {
        string principal, secundario;
        cin >> principal >> secundario;

        if (secundario.size() > principal.size()) {
            cout << "nao encaixa\n";
        } else {
            string finalPrincipal = principal.substr(principal.size() - secundario.size());

            if (finalPrincipal == secundario)
                cout << "encaixa\n";
            else
                cout << "nao encaixa\n";
        }
    }

    return 0;
}

File: 1241.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        while (n-- > 0) {
            String principal = sc.next();
            String secundario = sc.next();

            if (principal.endsWith(secundario))
                System.out.println("encaixa");
            else
                System.out.println("nao encaixa");
        }

        sc.close();
    }
}
